main.go

```diff
@@ -365,3 +522,83 @@ func removeAutoGoMod(m *moduleVCSInfo) (bool, error) {
 func autoGoMod(mpath string) string {
 	return "// Generated by gohack; DO NOT EDIT.\nmodule " + mpath + "\n"
 }
+
+func isEmptyDir(dir string) (bool, error) {
+	f, err := os.Open(dir)
+	if err != nil {
+		return false, errors.Wrap(err)
+	}
+	defer f.Close()
+	_, err = f.Readdir(1)
+	if err != nil && err != io.EOF {
+		return false, errors.Wrap(err)
+	}
+	return err == io.EOF, nil
+}
+
+func copyFile(dst, src string) error {
+	srcInfo, srcErr := os.Lstat(src)
+	if srcErr != nil {
+		return errors.Wrap(srcErr)
+	}
+	_, dstErr := os.Lstat(dst)
+	if dstErr == nil {
+		return errors.Newf("will not overwrite %q", dst)
+	}
+	if !os.IsNotExist(dstErr) {
+		return errors.Wrap(dstErr)
+	}
+	switch mode := srcInfo.Mode(); mode & os.ModeType {
+	case os.ModeSymlink:
+		return errors.Newf("will not copy symbolic link")
+	case os.ModeDir:
+		return copyDir(dst, src)
+	case 0:
+		return copyFile1(dst, src)
+	default:
+		return fmt.Errorf("cannot copy file with mode %v", mode)
+	}
+}
+
+func copyFile1(dst, src string) error {
```

Comment:
and then I'd actually call this `copyFile`.

Reply:
done